### ISSUE TITLE
libsched.a: No such file or directory while `createami`

### ISSUE THREAD

**Issue by sunbc0120** [aws/aws-parallelcluster#2934]
If you are reporting an issue with AWS Parallelcluster / CfnCluster please make sure to add the following data in order to facilitate the root cause detection:

**Required Info:**
 - AWS ParallelCluster version [e.g. 2.9.0]: 2.11.0
 - Full cluster configuration without any credentials or personal data
 - Cluster name: 
 - [Optional] Arn of the cluster CloudFormation main stack:

**Bug description and how to reproduce:**
A clear and concise description of what the bug is and the steps to reproduce the behavior.

Following the tutorial at: https://github.com/aws-samples/aws-parallelcluster-megatron/blob/main/TUTORIAL.md#building-an-custom-ami

While creating the AMI, it fails with:

<details>
<summary>log</summary>

```bash
...
custom-alinux2: a - cl_tcp_framework.o
custom-alinux2: a - cl_communication.o
custom-alinux2: a - cl_xml_parsing.o
custom-alinux2: a - cl_connection_list.o
custom-alinux2: a - cl_app_message_queue.o
custom-alinux2: a - cl_message_list.o
custom-alinux2: a - cl_fd_list.o
custom-alinux2: a - cl_host_list.o
custom-alinux2: a - cl_host_alias_list.o
custom-alinux2: a - cl_endpoint_list.o
custom-alinux2: a - cl_handle_list.o
custom-alinux2: a - cl_application_error_list.o
custom-alinux2: a - cl_commlib.o
custom-alinux2: a - cl_ssl_framework.o
custom-alinux2: echo libcomm.a
custom-alinux2: libcomm.a
custom-alinux2: cc -DSGE_ARCH_STRING=\"lx-amd64\" -O2 -Wstrict-prototypes -DLINUX -DLINUXAMD64 -DLINUXAMD64 -D_GNU_SOURCE -DGETHOSTBYNAME_R6 -DGETHOSTBYADDR_R8  -DTARGET_64BIT  -DSGE_PQS_API -DSPOOLING_dynamic -DSECURE -DHAVE_HWLOC=1 -DNO_JNI -DCOMPILE_DC -D__SGE_COMPILE_WITH_GETTEXT__  -D__SGE_NO_USERMAPPING__ -I../common -I../libs -I../libs/uti -I../libs/juti -I../libs/gdi -I../libs/japi -I../libs/sgeobj -I../libs/cull -I../libs/comm -I../libs/comm/lists -I../libs/sched -I../libs/evc -I../libs/evm -I../libs/mir -I../daemons/common -I../daemons/qmaster -I../daemons/execd -I../clients/common -I. -o test_sge_object -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_sge_object.o libsgeobj.a libsgeobjd.a libcull.a libcomm.a libcommlists.a libuti.a  -luti -ldl -lssl -lcrypto -lm -lpthread
custom-alinux2: cc -DSGE_ARCH_STRING=\"lx-amd64\" -O2 -Wstrict-prototypes -DLINUX -DLINUXAMD64 -DLINUXAMD64 -D_GNU_SOURCE -DGETHOSTBYNAME_R6 -DGETHOSTBYADDR_R8  -DTARGET_64BIT  -DSGE_PQS_API -DSPOOLING_dynamic -DSECURE -DHAVE_HWLOC=1 -DNO_JNI -DCOMPILE_DC -D__SGE_COMPILE_WITH_GETTEXT__  -D__SGE_NO_USERMAPPING__ -I../common -I../libs -I../libs/uti -I../libs/juti -I../libs/gdi -I../libs/japi -I../libs/sgeobj -I../libs/cull -I../libs/comm -I../libs/comm/lists -I../libs/sched -I../libs/evc -I../libs/evm -I../libs/mir -I../daemons/common -I../daemons/qmaster -I../daemons/execd -I../clients/common -I. -o test_sge_schedd_conf -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_sge_schedd_conf.o libsgeobj.a libsgeobjd.a libcull.a libcomm.a libcommlists.a libuti.a  -luti -ldl -lssl -lcrypto -lm -lpthread
custom-alinux2: cc -DSGE_ARCH_STRING=\"lx-amd64\" -O2 -Wstrict-prototypes -DLINUX -DLINUXAMD64 -DLINUXAMD64 -D_GNU_SOURCE -DGETHOSTBYNAME_R6 -DGETHOSTBYADDR_R8  -DTARGET_64BIT  -DSGE_PQS_API -DSPOOLING_dynamic -DSECURE -DHAVE_HWLOC=1 -DNO_JNI -DCOMPILE_DC -D__SGE_COMPILE_WITH_GETTEXT__  -D__SGE_NO_USERMAPPING__ -I../common -I../libs -I../libs/uti -I../libs/juti -I../libs/gdi -I../libs/japi -I../libs/sgeobj -I../libs/cull -I../libs/comm -I../libs/comm/lists -I../libs/sched -I../libs/evc -I../libs/evm -I../libs/mir -I../daemons/common -I../daemons/qmaster -I../daemons/execd -I../clients/common -I. -o test_sge_utility -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_sge_utility.o libsgeobj.a libsgeobjd.a libcull.a libcomm.a libcommlists.a libuti.a  -luti -ldl -lssl -lcrypto -lm -lpthread
custom-alinux2: cc -DSGE_ARCH_STRING=\"lx-amd64\" -O2 -Wstrict-prototypes -DLINUX -DLINUXAMD64 -DLINUXAMD64 -D_GNU_SOURCE -DGETHOSTBYNAME_R6 -DGETHOSTBYADDR_R8  -DTARGET_64BIT  -DSGE_PQS_API -DSPOOLING_dynamic -DSECURE -DHAVE_HWLOC=1 -DNO_JNI -DCOMPILE_DC -D__SGE_COMPILE_WITH_GETTEXT__  -D__SGE_NO_USERMAPPING__ -I../common -I../libs -I../libs/uti -I../libs/juti -I../libs/gdi -I../libs/japi -I../libs/sgeobj -I../libs/cull -I../libs/comm -I../libs/comm/lists -I../libs/sched -I../libs/evc -I../libs/evm -I../libs/mir -I../daemons/common -I../daemons/qmaster -I../daemons/execd -I../clients/common -I. -o test_sge_resource_quota -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_sge_resource_quota.o libsgeobj.a libsgeobjd.a libcull.a libcomm.a libcommlists.a libuti.a  -luti -ldl -lssl -lcrypto -lm -lpthread
custom-alinux2: cc -DSGE_ARCH_STRING=\"lx-amd64\" -O2 -Wstrict-prototypes -DLINUX -DLINUXAMD64 -DLINUXAMD64 -D_GNU_SOURCE -DGETHOSTBYNAME_R6 -DGETHOSTBYADDR_R8  -DTARGET_64BIT  -DSGE_PQS_API -DSPOOLING_dynamic -DSECURE -DHAVE_HWLOC=1 -DNO_JNI -DCOMPILE_DC -D__SGE_COMPILE_WITH_GETTEXT__  -D__SGE_NO_USERMAPPING__ -I../common -I../libs -I../libs/uti -I../libs/juti -I../libs/gdi -I../libs/japi -I../libs/sgeobj -I../libs/cull -I../libs/comm -I../libs/comm/lists -I../libs/sched -I../libs/evc -I../libs/evm -I../libs/mir -I../daemons/common -I../daemons/qmaster -I../daemons/execd -I../clients/common -I. -o test_eval_expression -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_eval_expression.o libsgeobj.a libsgeobjd.a libcull.a libcomm.a libcommlists.a libuti.a  -luti -ldl -lssl -lcrypto -lm -lpthread
custom-alinux2: cc -DSGE_ARCH_STRING=\"lx-amd64\" -O2 -Wstrict-prototypes -DLINUX -DLINUXAMD64 -DLINUXAMD64 -D_GNU_SOURCE -DGETHOSTBYNAME_R6 -DGETHOSTBYADDR_R8  -DTARGET_64BIT  -DSGE_PQS_API -DSPOOLING_dynamic -DSECURE -DHAVE_HWLOC=1 -DNO_JNI -DCOMPILE_DC -D__SGE_COMPILE_WITH_GETTEXT__  -D__SGE_NO_USERMAPPING__ -I../common -I../libs -I../libs/uti -I../libs/juti -I../libs/gdi -I../libs/japi -I../libs/sgeobj -I../libs/cull -I../libs/comm -I../libs/comm/lists -I../libs/sched -I../libs/evc -I../libs/evm -I../libs/mir -I../daemons/common -I../daemons/qmaster -I../daemons/execd -I../clients/common -I. -o test_sge_mirror -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_sge_mirror.o sig_handlers.o -lmir -levc -lgdi -lsgeobj -lsgeobjd -lcomm -lcommlists -lcull -luti  -lssl -lcrypto  -lm -lpthread -luti -ldl -lc
custom-alinux2: cc -o server_demo -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  server_demo.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o client_demo -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  client_demo.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o external_fd_demo -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  external_fd_demo.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o external_fd_server_demo -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  external_fd_server_demo.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_cl_commlib -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_cl_commlib.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_cl_commlib_client -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_cl_commlib_client.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_cl_hostname_list -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_cl_hostname_list.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_cl_hostalias_list -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_cl_hostalias_list.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_multithread_io -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_multithread_io.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_status_info -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_status_info.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_server_connection -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_server_connection.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_cl_endpoint_list -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_cl_endpoint_list.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_cl_com_parameter_list -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_cl_com_parameter_list.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_virtual_qmaster -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_virtual_qmaster.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_virtual_event_client -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_virtual_event_client.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_virtual_gdi_client -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_virtual_gdi_client.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_debug_client -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_debug_client.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_general -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_general.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_commlib_issue -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_commlib_issue.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_cl_issue_2747_server -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_cl_issue_2747_server.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_cl_issue_2747_client -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_cl_issue_2747_client.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_issues -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_issues.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o test_thread_throughput -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  test_thread_throughput.o libcomm.a -lssl -lcrypto libcommlists.a libuti.a  -luti -ldl -lm -lpthread
custom-alinux2: cc -o pdc -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  pdc_main.o procfs_main.o libsched.a libsgeobj.a libcomm.a libcommlists.a libsgeobjd.a libcull.a libuti.a libuti2.a  -luti -ldl -lssl -lcrypto  -lm -lpthread
custom-alinux2: cc -I../daemons/shepherd -DSGE_ARCH_STRING=\"lx-amd64\" -O2 -Wstrict-prototypes -DLINUX -DLINUXAMD64 -DLINUXAMD64 -D_GNU_SOURCE -DGETHOSTBYNAME_R6 -DGETHOSTBYADDR_R8  -DTARGET_64BIT  -DSGE_PQS_API -DSPOOLING_dynamic -DSECURE -DHAVE_HWLOC=1 -DNO_JNI -DCOMPILE_DC -D__SGE_COMPILE_WITH_GETTEXT__  -D__SGE_NO_USERMAPPING__ -I../common -I../libs -I../libs/uti -I../libs/juti -I../libs/gdi -I../libs/japi -I../libs/sgeobj -I../libs/cull -I../libs/comm -I../libs/comm/lists -I../libs/sched -I../libs/evc -I../libs/evm -I../libs/mir -I../daemons/common -I../daemons/qmaster -I../daemons/execd -I../clients/common -I. -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  -o sge_shepherd shepherd.o builtin_starter.o setrlimits.o signal_queue.o sge_shepconf.o setjoblimit.o sge_pset.o sge_fileio.o sge_shepherd_ijs.o shepherd_binding.o sge_ijs_comm.o sge_ijs_threads.o config_file.o err_trace.o execution_states.o qlogin_starter.o setosjobid.o sge_parse_num_par.o pdc.o procfs.o sge_mt_init.o sge_processes_irix.o sge_spool.o binding_support.o  -lgdi -lsgeobj -lsgeobjd  -lcull -lcomm -lcommlists -luti -luti2 -lhwloc  -luti -ldl -lssl -lcrypto  -lm -lpthread
custom-alinux2: cc -o qalter -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  qalter.o  sig_handlers.o parse_qsub.o show_job.o parse_job_cull.o usage.o sge_options.o read_defaults.o -lgdi -lsgeobj -lsgeobjd -lcull -lcomm -lcommlists -luti  -lssl -lcrypto  -luti -ldl  -lm -lpthread
custom-alinux2: cc -o qdel -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  qdel.o usage.o sge_options.o -lgdi -lsgeobj -lsgeobjd -lcull -lcomm -lcommlists -luti  -lssl -lcrypto  -luti -ldl  -lm -lpthread
custom-alinux2: cc -o qmod -L.  -rdynamic -Wl,-rpath,\$ORIGIN/../../lib/lx-amd64  qmod.o sig_handlers.o usage.o sge_options.o -lgdi -lsgeobj -lsgeobjd -lcull -lcomm -lcommlists -luti  -lssl -lcrypto  -luti -ldl  -lm -lpthread
custom-alinux2: ### libdrmaa.so.1.0
custom-alinux2: not done
custom-alinux2: STDERR: ../libs/uti/sge_unistd.c: In function ‘sge_rmdir’:
custom-alinux2: ../libs/uti/sge_unistd.c:452:4: warning: ‘readdir_r’ is deprecated [-Wdeprecated-declarations]
custom-alinux2:     while (SGE_READDIR_R(dir, (SGE_STRUCT_DIRENT *)dirent, &dent)==0 && dent!=NULL) {
custom-alinux2:     ^~~~~
custom-alinux2: In file included from ../libs/uti/sge_unistd.h:39:0,
custom-alinux2:                  from ../libs/uti/sge_unistd.c:42:
custom-alinux2: /usr/include/dirent.h:183:12: note: declared here
custom-alinux2:  extern int readdir_r (DIR *__restrict __dirp,
custom-alinux2:             ^~~~~~~~~
custom-alinux2: ../libs/spool/sge_dirent.c: In function ‘sge_get_dirents’:
custom-alinux2: ../libs/spool/sge_dirent.c:74:4: warning: ‘readdir_r’ is deprecated [-Wdeprecated-declarations]
custom-alinux2:     while (SGE_READDIR_R(cwd, (SGE_STRUCT_DIRENT *)dirent, &dent)==0 && dent!=NULL) {
custom-alinux2:     ^~~~~
custom-alinux2: In file included from ../libs/uti/sge_unistd.h:39:0,
custom-alinux2:                  from ../libs/spool/sge_dirent.c:41:
custom-alinux2: /usr/include/dirent.h:183:12: note: declared here
custom-alinux2:  extern int readdir_r (DIR *__restrict __dirp,
custom-alinux2:             ^~~~~~~~~
custom-alinux2: ../common/parse_qsub.c: In function ‘cull_parse_cmdline’:
custom-alinux2: ../common/parse_qsub.c:1133:18: warning: comparison between pointer and zero character constant [-Wpointer-compare]
custom-alinux2:           if (*sp == '\0') {
custom-alinux2:                   ^~
custom-alinux2: ../common/parse_qsub.c:1133:14: note: did you mean to dereference the pointer?
custom-alinux2:           if (*sp == '\0') {
custom-alinux2:               ^
custom-alinux2: ../libs/comm/test_cl_hostname_list.c: In function ‘main’:
custom-alinux2: ../libs/comm/test_cl_hostname_list.c:79:16: warning: iteration 2147483647 invokes undefined behavior [-Waggressive-loop-optimizations]
custom-alinux2:    while ((loops--) != 0) {
custom-alinux2:           ~~~~~~^~~
custom-alinux2: ../libs/comm/test_cl_hostname_list.c:79:9: note: within this loop
custom-alinux2:    while ((loops--) != 0) {
custom-alinux2:          ^
custom-alinux2: ../daemons/shepherd/builtin_starter.c: In function ‘son’:
custom-alinux2: ../daemons/shepherd/builtin_starter.c:574:10: warning: ignoring return value of ‘sge_chdir’, declared with attribute warn_unused_result [-Wunused-result]
custom-alinux2:           sge_chdir(pw->pw_dir);
custom-alinux2:           ^~~~~~~~~~~~~~~~~~~~~
custom-alinux2: cc: error: libsched.a: No such file or directory
custom-alinux2: make: *** [pdc] Error 1
custom-alinux2: make: *** Waiting for unfinished jobs....
custom-alinux2: ---- End output of "bash"  ----
custom-alinux2: Ran "bash"  returned 1
==> custom-alinux2: Provisioning step had errors: Running the cleanup provisioner, if present...
==> custom-alinux2: Terminating the source AWS instance...
==> custom-alinux2: Cleaning up any extra volumes...
==> custom-alinux2: No volumes to clean up, skipping
==> custom-alinux2: Deleting temporary security group...
==> custom-alinux2: Deleting temporary keypair...
Build 'custom-alinux2' errored after 18 minutes 9 seconds: Error executing Chef: Non-zero exit status: 1

==> Wait completed after 18 minutes 9 seconds

==> Some builds didn't complete successfully and had errors:
--> custom-alinux2: Error executing Chef: Non-zero exit status: 1

==> Builds finished but no artifacts were created.

```
</details>

**Additional context:**
Any other context about the problem. E.g.:
 - pre/post-install scripts, if any
 - screenshots, if useful

**Comment by lukeseawalker** [comment]
Hello @sunbc0120,
while I'm trying to reproduce this, can you please attach the full log so that I can see which recipe is failing? Seems like the SGE installation one, but I want to be sure I'm looking at the right place.

Thanks

**Comment by lukeseawalker** [comment]
I wasn't able to reproduce the error with the following:
```
$ DEEP_LEARNING_AMI_ID=`aws ec2 describe-images --owners amazon \
    --query 'Images[*].{ImageId:ImageId,CreationDate:CreationDate}' \
    --filters "Name=name,Values='Deep Learning AMI (Amazon Linux 2) Version 38.0'" \
    --region ${AWS_REGION} \
    | jq -r 'sort_by(.CreationDate)[-1] | .ImageId'`
    
$ AWS_REGION=us-east-1

$ wget https://raw.githubusercontent.com/aws-samples/aws-parallelcluster-megatron/main/scripts/custom_dlami_user_data.sh -O /tmp/custom_dlami_user_data.sh

$ pcluster createami -ai $DEEP_LEARNING_AMI_ID \  
        -os alinux2 \
        -ap megatron-on-pcluster- \
        -i p2.xlarge \
        --post-install file:///tmp/custom_dlami_user_data.sh \
        -r ${AWS_REGION}

Building AWS ParallelCluster AMI. This could take a while...
Base AMI ID: ami-00f44084952227ef0
Base AMI OS: alinux2
Instance Type: p2.xlarge
Region: us-east-1
VPC ID: xxx
Subnet ID: xxx
Template: https://us-east-1-aws-parallelcluster.s3.us-east-1.amazonaws.com/templates/aws-parallelcluster-2.11.0.cfn.json
Cookbook: https://us-east-1-aws-parallelcluster.s3.us-east-1.amazonaws.com/cookbooks/aws-parallelcluster-cookbook-2.11.0.tgz
Post install script url: file:///tmp/custom_dlami_user_data.sh
Post install script dir /var/folders/ln/jkgd88zs2rq03vq_nk69k5pmfp69xn/T/tmp4xuukcqo/script/20210714-155111-custom_dlami_user_data.sh
Packer log: /var/folders/ln/jkgd88zs2rq03vq_nk69k5pmfp69xn/T/packer.log.20210714-155111.22kcngwi
Packer Instance ID: i-0845001399663cf81
Packer status: 	exit code 0

Custom AMI ami-077538b4b4e9df547 created with name megatron-on-pcluster-aws-parallelcluster-2.11.0-amzn2-hvm-x86_64-202107141551

To use it, add the following variable to the AWS ParallelCluster config file, under the [cluster ...] section
```

The only difference I can spot is in the instance type I have used, I will try with p4d.

**Comment by sunbc0120** [comment]
> Hello @sunbc0120,
> while I'm trying to reproduce this, can you please attach the full log so that I can see which recipe is failing? Seems like the SGE installation one, but I want to be sure I'm looking at the right place.
> 
> Thanks

Thanks, @lukeseawalker , here is the full log:

[packer.log](https://github.com/aws/aws-parallelcluster/files/6819384/packer.log)

Yes, we used `p4d`.

**Comment by lukeseawalker** [comment]
Hello @sunbc0120,
I was able to reproduce. While I continue looking into this, the workaround for now is to use PCluster version `2.10.4`

Thanks

**Comment by lukeseawalker** [comment]
Hello @sunbc0120, this has been resolved in [version 2.11.1](https://github.com/aws/aws-parallelcluster/releases/tag/v2.11.1)
